### PR TITLE
Detailed error and warnings upon next() call

### LINKED ISSUE #13466: TypeError: Cannot read property 'isNextDevCommand' of undefined
# Bug report

## Describe the bug
TypeError: Cannot read property 'isNextDevCommand' of undefined options in createServer() do not seem to have the isNextDevCommand property

## To Reproduce
1. start custom server
2. navigate to /
3. See error

## Expected behavior
Not this

## Screenshots

Screen Shot 2020-05-27 at 12 56 41 PM

## System information

- OS: macOS
- Browser (if applies) chrome
- Version of Next.js: 9.3.2
- Version of Node.js: v12.16.2

### PR DESCRIPTION
**Pull request by todortotev**
**First, apologies for a second PR on the same issue but I was working on this already so I thought I'd push it and let you decide which you want to merge.**

The PR is related to [13466](https://github.com/vercel/next.js/issues/13466).

Based on my research, the error happens if the options are empty, null, or undefined. That's why I have decided that the most proper check would be using the! post-fix expression operator may assert that its operand is non-null and non-undefined. ``if (options == null)``

(Optional)
I have also added a warning, which warns the user if the passed "dev" variable is not a boolean.

It's my first PR on the "packages" part of the repo so I'd be glad to receive all kinds of critics. If you want me to change or remove anything, I'm open to suggestions.

---

Fixes #13466

### CODE REVIEW

File: packages/next/server/next.ts [hunk 1]
```diff
@@ -13,6 +13,12 @@ type NextServerConstructor = Omit<ServerConstructor, 'staticMarkup'> & {
 function createServer(options: NextServerConstructor): Server {
   const standardEnv = ['production', 'development', 'test']
 
+  if (options == null) {
```
Comment (timneutkens) [inline packages/next/server/next.ts:16]:
```suggestion
  if (typeof options === 'undefined') {
```

File: packages/next/server/next.ts [hunk 2]
```diff
@@ -22,6 +28,13 @@ function createServer(options: NextServerConstructor): Server {
   }
 
   if (options.dev) {
+    // This is optional and not related to the issue, but I believe this PR can introduce a handy warning if the variable is not boolean
```
Comment (timneutkens) [inline packages/next/server/next.ts:31]:
```suggestion
```

**Review by timneutkens — COMMENTED** [review]
Can you add tests for these 2 behaviors to `test/integration`, you can base it off of how `test/integration/cli` works

**Comment by TodorTotev** [comment]
@timneutkens I have added tests, but one thing that I wanted to note is that with 

```if (typeof options === 'undefined')```

 if I pass ```next(null)``` , the custom error won't trigger and instead, we're getting the ```"Cannot read property 'isNextDevCommand' of null"``` message again.

**Comment by Timer** [comment]
Let's switch it back to `== null`.

File: packages/next/server/next.ts [hunk 3]
```diff
@@ -13,6 +13,12 @@ type NextServerConstructor = Omit<ServerConstructor, 'staticMarkup'> & {
 function createServer(options: NextServerConstructor): Server {
   const standardEnv = ['production', 'development', 'test']
 
+  if (typeof options === 'undefined') {
```
Comment (Timer) [inline packages/next/server/next.ts:16]:
```suggestion
  if (options == null) {
```

File: test/integration/invalid-server-options/test/index.test.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    expect(() => next()).toThrowError(
+      'The server has not been instantiated properly. https://err.sh/next.js/invalid-server-options'
+    )
+  })
+
+  test('next() called with undefined parameters should throw error', () => {
+    expect(() => next(undefined)).toThrowError(
+      'The server has not been instantiated properly. https://err.sh/next.js/invalid-server-options'
+    )
+  })
```
Comment (Timer) [inline test/integration/invalid-server-options/test/index.test.js:18]:
```suggestion
  test('next() called with undefined parameter should throw error', () => {
    expect(() => next(undefined)).toThrowError(
      'The server has not been instantiated properly. https://err.sh/next.js/invalid-server-options'
    )
  })

  test('next() called with null parameter should throw error', () => {
    expect(() => next(null)).toThrowError(
      'The server has not been instantiated properly. https://err.sh/next.js/invalid-server-options'
    )
  })
```

**Review by Timer — CHANGES_REQUESTED** [review]

**Review by Timer — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| buildDuration | 10.1s | 10.8s | ⚠️ +630ms |
| nodeModulesSize | 66 MB | 66 MB | ⚠️ +926 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.815 | 1.741 | -0.07 |
| / avg req/sec | 1377.1 | 1436.06 | ⚠️ +58.96 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.134 | 1.203 | ⚠️ +0.07 |
| /error-in-render avg req/sec | 2204.74 | 2077.95 | -126.79 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..6c2a.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.21 kB | 8.21 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.51 kB | 1.51 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.32 kB | 5.32 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| index.html gzip | 924 B | 924 B | ✓ |
| link.html gzip | 934 B | 934 B | ✓ |
| withRouter.html gzip | 922 B | 922 B | ✓ |
| Overall change | 2.78 kB | 2.78 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| buildDuration | 10.9s | 11s | ⚠️ +104ms |
| nodeModulesSize | 66 MB | 66 MB | ⚠️ +926 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..6c2a.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.21 kB | 8.21 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.51 kB | 1.51 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.32 kB | 5.32 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _error.js | 878 kB | 878 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 879 kB | 879 kB | ✓ |
| link.js | 916 kB | 916 kB | ✓ |
| routerDirect.js | 909 kB | 909 kB | ✓ |
| withRouter.js | 909 kB | 909 kB | ✓ |
| Overall change | 4.5 MB | 4.5 MB | ✓ |

</details>

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| buildDuration | 12.2s | 12s | -193ms |
| nodeModulesSize | 66 MB | 66 MB | ⚠️ +883 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.198 | 2.14 | -0.06 |
| / avg req/sec | 1137.56 | 1168.16 | ⚠️ +30.6 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.407 | 1.378 | -0.03 |
| /error-in-render avg req/sec | 1776.47 | 1814.4 | ⚠️ +37.93 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..6c2a.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.21 kB | 8.21 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.51 kB | 1.51 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.32 kB | 5.32 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| index.html gzip | 924 B | 924 B | ✓ |
| link.html gzip | 934 B | 934 B | ✓ |
| withRouter.html gzip | 922 B | 922 B | ✓ |
| Overall change | 2.78 kB | 2.78 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| buildDuration | 12.9s | 12.8s | -166ms |
| nodeModulesSize | 66 MB | 66 MB | ⚠️ +883 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..6c2a.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.21 kB | 8.21 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.51 kB | 1.51 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.32 kB | 5.32 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _error.js | 878 kB | 878 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 879 kB | 879 kB | ✓ |
| link.js | 916 kB | 916 kB | ✓ |
| routerDirect.js | 909 kB | 909 kB | ✓ |
| withRouter.js | 909 kB | 909 kB | ✓ |
| Overall change | 4.5 MB | 4.5 MB | ✓ |

</details>

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| buildDuration | 11.9s | 11.9s | ⚠️ +11ms |
| nodeModulesSize | 66 MB | 66 MB | ⚠️ +883 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.164 | 2.037 | -0.13 |
| / avg req/sec | 1155.28 | 1227.21 | ⚠️ +71.93 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.176 | 1.135 | -0.04 |
| /error-in-render avg req/sec | 2125.64 | 2201.95 | ⚠️ +76.31 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..6c2a.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.21 kB | 8.21 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.51 kB | 1.51 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.32 kB | 5.32 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| index.html gzip | 924 B | 924 B | ✓ |
| link.html gzip | 934 B | 934 B | ✓ |
| withRouter.html gzip | 922 B | 922 B | ✓ |
| Overall change | 2.78 kB | 2.78 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| buildDuration | 13.4s | 13s | -382ms |
| nodeModulesSize | 66 MB | 66 MB | ⚠️ +883 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..6c2a.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.21 kB | 8.21 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.51 kB | 1.51 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.32 kB | 5.32 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js proper-server-error | Change |
| - | - | - | - |
| _error.js | 878 kB | 878 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 879 kB | 879 kB | ✓ |
| link.js | 916 kB | 916 kB | ✓ |
| routerDirect.js | 909 kB | 909 kB | ✓ |
| withRouter.js | 909 kB | 909 kB | ✓ |
| Overall change | 4.5 MB | 4.5 MB | ✓ |

</details>

</details>

**Comment by TodorTotev** [comment]
@Timer Im glad you liked my == null proposal :D

**Pull request merged**